### ISSUE TITLE
Bastion not accessible over SSH from inside EC2

### ISSUE THREAD

**Issue by ahawkins** [kubernetes/kops#1563]
**Kops Version**: 1.5.0-alpha3

I created a Kops cluster with this command (this from an ansible playbook, but that shouldn't make a difference.) The `{{ }}` are variables filled in by ansible:

```
      command: >
        kops create cluster
          --yes
          --bastion
          --topology=private
          --networking=weave
          --associate-public-ip=false
          --node-count=3
          --cloud=aws
          --zones={{ item.zones | join(',') }}
          --ssh-public-key=~/k8s_node_key.pub
          --dns-zone={{ production_tld }}
          --name={{ item.name }}
```

I can access the bastion over ssh from my laptop. However I cannot ssh into the bastion from inside EC2. I found this running my ansible playbook. The playbook runs from a dedicated machine in EC2 to speed up all the AWS api calls.

I fixed this by adding a ingress rule on port 22 from `0.0.0.0/0`. I _think_ the problem is the ingress rule for the bastion security group. The security group only allows traffic on port 22 from _itself_. I've never seen this before and cannot find documentation on the impact. 

**Proposal**: Use `0.0.0.0/0` ingress rule so one can SSH into the bastion from anywhere on the internet. That is the point after all right? :)

**Comment by justinsb** [comment]
You're quite right, I believe.  Related to the work happening in https://github.com/kubernetes/kops/pull/1527/files

**Comment by ahawkins** [comment]
@justinsb lovely. You can ping me or @miyurz to test a fix for this.

**Comment by AlexLast** [comment]
@ahawkins If you use the --admin-access flag you can set this.. e.g

```--admin-access 5X.2XX.2XX.XX/32```

^ The above being your EC2 instance. I think a proposal to allow 0.0.0.0/0 by default on port 22 is pretty insecure.

**Comment by ahawkins** [comment]
>  I think a proposal to allow 0.0.0.0/0 by default on port 22 is pretty insecure.

@AlexLast my proposal is to allow traffic on port 22 from 0.0.0.0/0 _only_ on the instance specifically designed for SSH access; not to k8s nodes.

**Comment by AlexLast** [comment]
@ahawkins I know, but the master/node security groups allow SSH access from the bastion.

**Comment by chrislovecnm** [comment]
Is this a bug?

**Comment by justinsb** [comment]
So I dug into this in more detail.  Here's how it works in 1.5.0 beta1:

* We create a SG for bastion instances.  It is names `bastion.<clustername>` and allows access over port 22 from `bastion-elb.<clustername>`.
* `bastion-elb.<clustername>` is the name for the security group on the ELB.  It allows access from sshAccess CIDRs.
* We set up a DNS name `bastion.<clustername>` which points to the ELB.  This is how you can access the bastion(s).

My guess is that you were trying to connect to the bastion by instance IP, or maybe had set a permissions policy that excluded your internal Amazon instance.

I think the sshAccess policy does work today, if you access the bastion by name through the ELB.  (We do that because otherwise we have no easy way to discover the bastion).

If this isn't right, or doesn't work for your use case, can you provide more details.  And sorry to flip-flop!

**Comment by ahawkins** [comment]
Seems this fixed on the latest 1.5. Created a new cluster from scratch and can ssh into the bastions from inside EC2 and the public internet.
